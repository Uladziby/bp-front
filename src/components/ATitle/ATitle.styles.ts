/** @format */

import styled from "styled-components";

export const StyledTitle = styled.h2<{
	size: number;
	weight: number;
	lineheight: number;
	marginBottom?: number;
	color?: string;
}>`
	font-size: ${({ size }) => size}px;
	font-weight: ${({ weight }) => weight};
	line-height: ${({ lineheight }) => lineheight}px;
	margin-bottom: ${({ marginBottom }) => marginBottom}px;
	color: ${({ color }) => color};
`;
