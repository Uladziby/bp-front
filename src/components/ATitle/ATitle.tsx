import { StyledTitle } from './ATitle.styles';
import { IATitleProps } from './type';

export const ATitle = ({
  size,
  weight,
  lineHeight,
  children,
  color,
  marginBottom,
  dataTestId,
  className,
}: IATitleProps) => (
  <StyledTitle
    size={size}
    weight={weight}
    lineHeight={lineHeight}
    color={color}
    marginBottom={marginBottom}
    data-testid={dataTestId}
    className={className}
  >
    {children}
  </StyledTitle>
);
