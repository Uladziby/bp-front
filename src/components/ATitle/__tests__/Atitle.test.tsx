import { RenderResult, render, screen } from '@testing-library/react';
import { ATitle } from '../ATitle';

describe('ATitle', () => {
  const props = {
    size: 30,
    weight: 900,
    lineHeight: 2,
    color: 'white',
    marginBottom: 20,
    children: <p>child</p>,
  };
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<ATitle {...props} />);
  });

  it('should render correctly', () => {
    const element = screen.getByRole('heading');

    expect(element).toBeInTheDocument();
    expect(element.innerHTML).toEqual('<p>child</p>');
  });

  it('snapshot', () => {
    const element = screen.getByRole('heading');

    expect(element).toMatchSnapshot();
  });

  it('handles property color correctly', () => {
    const element = screen.getByRole('heading');

    expect(element).toHaveStyle({ color: 'white' });
  });

  it('handles property marginBottom correctly', () => {
    const element = screen.getByRole('heading');

    expect(element).toHaveStyle({ marginBottom: '20px' });
  });

  it('handles property size correctly', () => {
    const element = screen.getByRole('heading');

    expect(element).toHaveStyle({ fontSize: '30px' });
  });

  it('handles property weight correctly', () => {
    const element = screen.getByRole('heading');

    expect(element).toHaveStyle({ fontWeight: '900' });
  });
});
