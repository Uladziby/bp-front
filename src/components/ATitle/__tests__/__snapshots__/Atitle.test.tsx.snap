// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATitle snapshot 1`] = `
<h2
  class="sc-bcXHqe eFrToj"
  color="white"
  size="30"
>
  <p>
    child
  </p>
</h2>
`;
